Cpp/fostgres/updater.cpp

```diff
@@ -106,7 +107,8 @@ std::pair<boost::shared_ptr<fostlib::mime>, int> fostgres::updater::insert(
 std::pair<boost::shared_ptr<fostlib::mime>, int> fostgres::updater::update(
         fostgres::updater::intermediate_data d,
         std::optional<std::size_t> row) {
-    cnx.update(relation.c_str(), d.first, d.second);
+    auto rel = relation;
+    cnx.update(rel.shrink_to_fit(), d.first, d.second);
```

Comment:
Why don't just `relation.shrink_to_fit()`?

Reply:
Because `rel` is `const` and we need a mutable object to call `shrink_to_fit` on.